src/test/java/seedu/address/model/person/MatriculationNumberTest.java

```diff
@@ -0,0 +1,42 @@
+package seedu.address.model.person;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import org.junit.jupiter.api.Test;
+
+public class MatriculationNumberTest {
+
+    @Test
+    public void constructor_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> new MatriculationNumber(null));
+    }
+
+    @Test
+    public void constructor_invalidMatriculationNumber_throwsIllegalArgumentException() {
+        String invalidMatriculationNumber = "";
+        assertThrows(IllegalArgumentException.class, () -> new MatriculationNumber(invalidMatriculationNumber));
+    }
+
+    @Test
+    public void isValidMatriculationNumber() {
+        // null matriculation number
+        assertThrows(NullPointerException.class, () -> MatriculationNumber.isValidMatriculationNumber(null));
+
+        // invalid matriculation number
```

Comment:
Probably can add test case when the first character is not an A

Reply:
and also trailing spaces

Reply:
I added the test case for the first character not being an a/A. Good eye! Thanks for spotting this out 👍